src/iminuit/minuit.py

```diff
@@ -1224,18 +1235,29 @@ def __call__(self, par, v):
             else:
                 method = "BFGS"
 
+        # copy to avoid mutable problems
+        scipy_options = dict(options)
+
+        # attempt to set default number of function evaluations if not provided
         # various workarounds for API inconsistencies in scipy.optimize.minimize
-        options = {"maxiter": ncall}
+        if "maxiter" not in options:
+            scipy_options["maxiter"] = ncall
         if method in (
             "Nelder-Mead",
             "Powell",
         ):
-            options["maxfev"] = ncall
-            del options["maxiter"]
+            if "maxfev" not in options:
+                scipy_options["maxfev"] = ncall
+
+            if "maxiter" not in options:
+                del scipy_options["maxiter"]
 
         if method in ("L-BFGS-B", "TNC"):
-            options["maxfun"] = ncall
-            del options["maxiter"]
+            if "maxfun" not in options:
+                scipy_options["maxfun"] = ncall
+
+            if "maxiter" not in options:
+                del scipy_options["maxiter"]
```

Comment:
Are these changes an improvement? I think it makes more sense to uniformly set a stopping criterion with ncall. I don't see how overriding this feature with option is helpful.

Reply:
Ok, after thinking more about it, I take this back. My first impression was that "maxiter", "maxfev", "maxfun" in options should not override ncall, but your solution is ok: if these options are explicitly set, then they override ncall, but if they are not set, ncall is used accordingly. options is for experts, so when you use it you are expected to know what you are doing, and it would be more surprising if setting maxiter etc wouldn't have the desired effect.